Makefile

```diff
@@ -40,8 +89,25 @@ test-coverage:
 	echo "" > $(COVERAGE_REPORT); \
 	$(GOTEST) -coverprofile=$(COVERAGE_REPORT) -coverpkg=./... -covermode=$(COVERAGE_MODE) ./...
 
+.PHONY: clean clean-docker clean-all
+
 clean:
-	rm -rf $(GIT_DIST_PATH)
+	@if [ -d $(GIT_DIST_PATH) ]; then
+		rm -rf $(GIT_DIST_PATH)
+	fi
+	if [ -f $(TEST_ENV) ] || [ -f $(DOCKER_ENV) ]; then
+		rm $(TEST_ENV) $(DOCKER_ENV)
+	fi
+	rm coverage*
+
+# doc sys prune in this case only removes dangling images, but filter it anyway
+GOGIT_IMAGES := $(shell docker image ls --filter reference=go-git -q)
+clean-docker:
+	docker system prune -f --filter "label=$(DOCKER_GOGIT_KEY)=$(DOCKER_GOGIT_LABEL)"
+	docker volume prune -f --filter "label=$(DOCKER_GOGIT_KEY)=$(DOCKER_GOGIT_LABEL)"
+	docker image rm -f $(GOGIT_IMAGES)
```

Comment:
For simplicity, we probably could just document this instead of adding an additional target. Although your implementation added filters, this is still a system-wide op.

Reply:
Okay, will do